src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -154,7 +154,11 @@ public static String getParentPath(String remotePath) {
     public static OCFile fillOCFile(RemoteFile remote) {
         OCFile file = new OCFile(remote.getRemotePath());
         file.setCreationTimestamp(remote.getCreationTimestamp());
-        file.setFileLength(remote.getLength());
+        if (remote.getMimeType().equalsIgnoreCase("DIR")){
+            file.setFileLength(remote.getSize());
```

Comment:
Ah, I almost missed this part, that's why I thought it couldn't be working.

@PVince81 , do you recall what's the minimum server version including the property ```oc:size``` ?